src/components/Authentication/Login.jsx

```diff
@@ -0,0 +1,46 @@
+import { useLocation, useNavigate } from "react-router";
+import { loginHandler } from "../../utils";
+import {useLogin} from "../../context"
+const Login = () => {
+
+    const {loginState,loginDispatch} = useLogin();
+    const location = useLocation();
+    const navigate = useNavigate();
+
+    const showPasswordHandler = () => loginState.showPassword ? loginDispatch({showPassword: false}):loginDispatch({showPassword: true});
```

Comment:
Agreed. Instead, you could use a local state for hiding/ showing the password.

Reply:
I thought since all the states are combined in the reducer I combined this as well instead of mixing useState and useReducer